ops/model.py

```diff
@@ -1039,20 +1039,36 @@ class Storage:
 
     Attributes:
         name: Simple string name of the storage
-        id: The provider id for storage
+        id: The index number for storage
     """
 
-    def __init__(self, storage_name, storage_id, backend):
+    def __init__(self, storage_name: str, storage_index: int, backend):
         self.name = storage_name
-        self.id = storage_id
+        self._index = storage_index
         self._backend = backend
         self._location = None
 
+    @property
+    def index(self) -> int:
+        """The index associated with the storage (usually 0 for singular storage)."""
+        return self._index
+
+    @property
+    def id(self) -> int:
+        """DEPRECATED (use ".index"): The index associated with the storage."""
+        logger.warning("model.Storage.id is being replaced - please use model.Storage.index")
+        return self.index
+
+    @property
+    def full_id(self) -> str:
```

Comment:
`full_name`?  and I know `identifier` is long for `id`, but if the long-term plan is to replace what used to be 'id' with what now is 'full_id' (so I thought), we could do:

old version:
id: X

current version:
id: X (deprecated: use self.index)
identifier: Y 
index: X

future version:
identifier: Y (deprecated: use id)
id: Y
index: X

future future version:
id: Y
index: X


Otherwise we go immediately for a hard replace of the name `id`, and wait for a major version bump to put this in.

Reply:
either way this is a non-blocking nit, feel free to ignore it

Reply:
I think I'm going to stick with `full_id`.  It's (somewhat) descriptive and retains the `id` suffix which helps the connection between this and `storage_id` be more obvious to users of our apis.

Reply:
+1